### ISSUE TITLE
(Feature Request): Support for m.youtube.com

### ISSUE THREAD

**Issue by Cyan903** [Anarios/return-youtube-dislike#477]
### Extension or Userscript?

Extension

### Request or suggest a new feature!

A lot of people (including me) prefer using [m.youtube.com](m.youtube.com) instead of the actual app. While options like [Newpipe](https://newpipe.net/) and [Vanced](https://vancedapp.com/) exist, I still prefer the website because I get to run all my favorite browser extensions like uBlock and Dark Reader. I also don't need to install another app, which is a bonus especially since YouTube comes pre-installed on most Android devices now. 

While it is possible to install this extension on certain mobile browsers (#228), [m.youtube.com](m.youtube.com) does not properly show the dislikes. So the only way to restore dislikes is to use desktop site, which is very annoying to use on mobile. I think adding support for [m.youtube.com](m.youtube.com) would go a long way for all the users who prefer using their web browsers instead of a modified app/alternative client.

### Ways to implement this!

There are two ways I can think of implementing this. If you're willing to jump through some hoops (https://github.com/Anarios/return-youtube-dislike/issues/228) to get desktop extensions running on mobile browsers, you can get this extension to run on [m.youtube.com](m.youtube.com). The problem is, it is very buggy. It doesn't always show the dislike count. Sometimes you have to reload multiple times. If it does manage to show the dislike count, it will also create other elements which shouldn't be visible. And if you click on any other videos, the dislike counter will not show until you reload.

### Example (highlighted in red)
![example](https://user-images.githubusercontent.com/88910588/154004215-8bbe9cab-6e23-4767-8fd9-de8aab472ee1.png)

### Method 1
So we can patch the current extension to work with [m.youtube.com](m.youtube.com). We would also have to make sure the extension is accessible to mobile browsers, as you cannot currently install the extension without relying on 3rd party software or tricks. This would be the fastest and easiest method.

### Method 2
The second option is to create an entirely new extension for mobile devices. This way we wouldn't have to worry about making the main extension compatible with mobile devices. Doing this would also keep the main extension less cluttered as a lot of the logic for this extension would never execute for desktop users, keeping both extensions smaller and easier to maintain.

Personally I think method 2 is a better choice, but I am willing to work on both. This is my first time making a Github issue, so please tell me if I'm doing anything wrong.

### Can you work on this?

- [X] Yes
- [ ] No

**Comment by bigbiggie** [comment]
Having the same issue using a mobile phone where the Youtube app isn't available, but the userscript works well for me. Do I miss something, when using the userscript instead of the extension? Could you please tell me, why you want the extension to work on the mobile version instead of using the userscript? Thanks.

**Comment by CyanPiano** [comment]
@bigbiggie Could you tell me what browser and extension you are using? I'm using [fennec](https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/) which doesn't let me run userscripts.

**Comment by sy-b** [comment]
Try this userscript https://github.com/Anarios/return-youtube-dislike/raw/856ebf11244a81e4c7675b2676ef04160e054af4/Extensions/UserScript/Return%20Youtube%20Dislike.user.js

**Comment by UglyDonkey** [comment]
I like more method 1 as it would be much easier to maintain one extension instead of keeping both of them in sync. In addition both versions of yt can be used on desktop and mobile browsers.

I think it can be easily achieved with creating two content scripts for both versions of yt which will implements connectors to page and third one would contain rest of the common logic which interacts with page over connectors. Then it only needs in manifests to specify content scripts as
`*://www.youtube.com/* - "content.desktop.js", "content.common.js"`
`*://m.youtube.com/* - "content.mobile.js", "content.common.js"`

This way main extension logic is not cluttered with details of page at all and should make it easy to keep compatibility with desktop and mobile.

**Comment by philipbrembeck** [comment]
Could you explain any further how to reproduce the misplaced dislike counts? It works fine for me.
<img width="374" alt="Bildschirmfoto 2022-02-21 um 10 17 50" src="https://user-images.githubusercontent.com/4144601/154925039-a1316634-bb75-49f8-a561-c89266ecb751.png">

**Comment by bigbiggie** [comment]
> @bigbiggie Could you tell me what browser and extension you are using? I'm using [fennec](https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/) which doesn't let me run userscripts.

Sorry, I don't use android. I didn't know that android firefox doesn't support userscripts. I'm using firefox desktop with a mobile patch running on Arch Linux Arm (Pinephone). Tampermonkey is my userscript manager addon.

**Comment by CyanPiano** [comment]
@philipbrembeck are you using the script or the extension? I don't think this will be very helpful, but my exact steps were to download the extension, visit youtube and click on a video. 

https://user-images.githubusercontent.com/88910588/155251687-24d0f50a-7768-4b12-97f6-93981ddda49e.mp4

I recreated this on my computer because it was much easier to record and upload, but the steps on mobile are the same. The browsers I've been able to replicate this are:

- Firefox Desktop 96.0.3 (using a user agent switcher to access m.youtube.com)
- Chromium 98.0.4758.80 (also using a user agent switcher)
- Kiwi Browser on Android 9.8.0.4758.106

What browser are you using?
